rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -1634,6 +1634,12 @@ static std::string make_fqtopic(
   return make_fqtopic(prefix, topic_name, suffix, qos_policies->avoid_ros_namespace_conventions);
 }
 
+static bool is_rmw_duration_default(rmw_time_t duration)
+{
+  return rmw_time_equal(duration, RMW_QOS_DURATION_UNSPECIFIED) ||
+         rmw_time_equal(duration, RMW_DURATION_INFINITE);
```

Comment:
I find it surprising that ∞ is treated as default as well. It does mean that `rmw_time_total_nsec` can be used instead of a Cyclone-specific conversion function mapping `RMW_DURATION_INFINITE` to `DDS_INFINITY`, but to me it is a bit of a stain on an otherwise clean proposal 🙂

Reply:
I agree with this comment.
Isn't possible to have another function to compare the input value to "RMW_DURATION_INFINITE" and in that case set `DDS_INFINITY`?

Reply:
Sure, that seems fine to me. I wasn't sure if setting e.g. Deadline to DDS_INFINITY was the same thing as not setting it at all. 

I noticed that in the current implementation, an input of 0 for `liveliness.lease_duration` is explicitly translated to `DDS_INFINITY` - but for Deadline and Lifespan, this same input is ignored, so the value is not set at all.

I would like for users providing `RMW_DURATION_INFINITE` to Deadline+Lifespan to get the same behavior as they do when passing 0 now. Does setting Deadline/Lifespan to `DDS_INFINITY` have the same behavior as not setting it?

Reply:
There is a small difference between setting it to ∞ and leaving it unset in where the value comes from: in Cyclone's C API, any QoS not set explicitly is inherited from the topic for those settings that also apply to the topic. The topics here do not override any defaults, so the default value of ∞ applies either way.

Ignoring a lease duration of 0 would work, too, for the same reasons. I suppose the difference is that deadline and lifespan can be set separately, but liveliness kind and lease duration can only be set together. So while one could ignore (automatic, 0), but any other liveliness setting requires setting something. And so one might as well do it always.

I can imagine that setting these defaults in all cases would probably avoid some head scratching in the future.

Reply:
In other rmw implementations (I'm not sure if here), the "default" can be modified externally in a QoS profile file, so it might be different to INFINITY.

> I can imagine that setting these defaults in all cases would probably avoid some head scratching in the future.

:+1:

Reply:
This makes sense, thanks! I will update the RMW API proposal to `0 == unspecified`, and `RMW_DURATION_INFINITE == DDS_INFINITY` - then update this PR to match.

Reply:
Ok - updated to treat the cases differently.